### PR TITLE
added gatk4 variantfiltration

### PR DESCRIPTION
**Pull request by KevinMenden**
Adding the GATK4 variantfiltration module.

This just takes a VCF files and filters it - the commands to filter must be given via `modules.config` (see https://gatk.broadinstitute.org/hc/en-us/articles/360050815032-VariantFiltration)

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: software/gatk4/variantfiltration/main.nf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        container "quay.io/biocontainers/gatk4:4.2.0.0--0"
+    }
+
+    input:
+    tuple val(meta), path(vcf)
+    tuple path(fasta), path(fai), path(dict)
+
+    output:
+    tuple val(meta), path("*filtered.vcf"), emit: vcf
+    path "*.version.txt"          , emit: version
```
Comment (drpatelh) [inline software/gatk4/variantfiltration/main.nf:27]:
```suggestion
    tuple val(meta), path("*filtered.vcf"), emit: vcf
    path "*.version.txt"                  , emit: version
```

Comment (drpatelh) [reply software/gatk4/variantfiltration/main.nf:27]: Seriously! Is it only me that sees this 😆

File: software/gatk4/variantfiltration/meta.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        e.g. [ id:'test']
+  - vcf:
+      type: list
+      description: Input VCF file
+      pattern: "*.{vcf}"
+  - fasta:
+      type: tuple of files
+      description: |
+        Tuple of fasta file (first), sequence dict (second) and fasta index (third)
+      pattern: ["*.fasta", "*.dict", "*.fai"]
```
Comment (drpatelh) [inline software/gatk4/variantfiltration/meta.yml:50]: Need to add entries here for `fai` and `dict`! Man we need `nf-core lint`!

**Review by drpatelh — APPROVED** [review]
Couple of minor changes @KevinMenden ! Thanks!

File: software/gatk4/variantfiltration/main.nf (hunk 1 above)
Comment (KevinMenden) [reply software/gatk4/variantfiltration/main.nf:27]: 👀

File: software/gatk4/variantfiltration/meta.yml (hunk 2 above)
Comment (KevinMenden) [reply software/gatk4/variantfiltration/meta.yml:50]: True - this is actually a good test, let's check whether the linter captures this

**Pull request merged**